packages/truffle-box/lib/utils/unbox.js

```diff
@@ -81,13 +66,36 @@ function fetchRepository(url, dir) {
   });
 }
 
-function copyTempIntoDestination(tmpDir, destination) {
-  return new Promise(function(accept, reject) {
-    fs.copy(tmpDir, destination, function(err) {
-      if (err) return reject(err);
-      accept();
-    });
-  });
+async function copyTempIntoDestination(tmpDir, destination, force) {
+  const currentContent = fs.readdirSync(destination);
+  if (!force) {
+    const tmpContent = fs.readdirSync(tmpDir);
+    for (let file of tmpContent) {
+      if (currentContent.includes(file)) {
+        console.log(`${file} already exists in this directory...`);
+        var overwriting = [
+          {
+            type: 'confirm',
+            name: 'overwrite',
+            message: `Overwrite ${file}?`,
+            default: false
+          }
+        ];
+
+        await inquirer.prompt(overwriting)
+          .then(answer => {
+            if (answer.overwrite) {
+              fs.removeSync(file);
+              fs.copySync(`${tmpDir}/${file}`, `${destination}/${file}`);
+            }
+          });
+      } else {
+        fs.copySync(`${tmpDir}/${file}`, `${destination}/${file}`);
+      }
+    }
+  } else {
+    fs.copySync(tmpDir, destination);
+  }
```

Comment:
General thoughts: this function seems to be outgrowing itself. I recommend re-organizing this by splitting it into multiple functions, or at least decreasing the number of levels of nesting.

I think you might be able to use `.filter()` to assign discrete ideas into specific variables. Possible approach, feel free to use as a starting point or not:

```javascript
const boxContents = fs.readdirSync(tmpDir);
const destinationContents = fs.readdirSync(destination);

const newContents = boxContents.filter(
  (filename) => !destinationContents.includes(filename)
);

const contentCollisions = boxContents.filter(
  (filename) => destinationContents.includes(filename)
);

let shouldCopy;
if (force) {
  // write all box contents
  shouldCopy = boxContents;
} else {
  // promptOverwrites returns list of files that the user says "yes" to overwrite
  const overwriteContents = await promptOverwrites(contentCollisions);
  shouldCopy = [...newContents, ...overwriteContents];
}

for (let file of shouldCopy) {
  fs.copySync(tmpDir, destination);
}